### ISSUE TITLE
dropping through case statements is NOT an error. Always

### ISSUE THREAD

**Issue by dexdyne** [koalaman/shellcheck#598]
I have a case statement which intentionally drops through.

This should create a warning, nott an error . 

I'd go even further - if the preceding line is a comment that includes either of the words "drop" or "though" it should be passed without comment :-)

**Comment by bittorf** [comment]
please give an example which we can copy/paste to shellcheck.net

**Comment by brother** [comment]
```shell
apa=foo

case $apa in
  foo)
       echo "first"
       # DROP TO NEXT INTENTIONAL
  bar)
       echo "second"
       ;;
esac
```

This produces [SC1009](https://github.com/koalaman/shellcheck/wiki/SC1009) for the case line. [SC1073](https://github.com/koalaman/shellcheck/wiki/SC1073) for "foo)". [SC1072](https://github.com/koalaman/shellcheck/wiki/SC1072) and [SC1074](https://github.com/koalaman/shellcheck/wiki/SC1074) for "bar)".

I have no opinion on what type of message to produce (warning/error). However, having a "fall through if magic keyword in comment" I am again

**Comment by dexdyne** [comment]
Well you don't have to put a comment there - if you don't you get the behaviour you prefer :-))

Anyway thanks for the example.

**Comment by bittorf** [comment]
i did not even know that - is it a bash extension?
it's invalid for POSIX.

**Comment by brother** [comment]
bash added the support "recent". But I am not sure my example above is for a correct implementation, it works but that's all.

The switch statement page at wikipedia has the following quote (without source): "Similarly, bash defaults to not falling through, when terminated with ;; but more recent versions allow fallthrough with ;& instead."

testing it in bash 4.3.42 I see that my example is broken.

```shell
#!/bin/bash
apa=foo

case $apa in
	foo)
		echo "first"
		# DROP TO NEXT INTENTIONAL
		;&
	bar)
		echo "second"
		;;
esac
```

There. And that code does NOT trigger any warnings. Maybe it's a case of broken code that shellcheck actually catches for OP?

**Comment by dexdyne** [comment]
Ah, I see this is probably my fault.
I had assumed, rather stupidly, that if you omit the ;; you naturally get fall-through, just like most languages.
I will go away and read about it some more.
My apologies!

**Comment by bittorf** [comment]
please close the issue...
